ueberdosis/tiptap
# Fix : React DND broken when used within a NodeViewWrapper

Drag event doesn't work on components wrapped inside a NodeViewWrapper.

## Please describe your changes

Currently, drag events are explicitly prevented on NodeView. This is also applied to NodeView children due to event propagation, thus preventing drag events on the components inside NodeView. This PR adds a check to ensure drag events are only prevented on the NodeView parent.

## How did you accomplish your changes

Added event target check to ensure drag prevention on NodeView doesn't propagate to child components.

```
    if (!isDraggable && isSelectable && isDragEvent && event.target === this.dom) {
      event.preventDefault()
    }

    if (isDraggable && isDragEvent && !isDragging && event.target === this.dom) {
      event.preventDefault()
      return false
    }
```

## How have you tested your changes

Created a NodeView with a react dnd component using NodeViewWrapper. Drag works as expected on the react dnd component.

## How can we verify your changes

Create a NodeView with draggable component and check if component can be dragged.

## Remarks

[add any additional remarks here]

## Checklist


## Related issues

#5018
